src/server/console/logind_console_services.cpp

```diff
@@ -229,17 +232,45 @@ connect_to_system_bus(mir::GLibMainLoop& ml)
             {
                 auto error_msg = error ? error->message : "unknown error";
                 BOOST_THROW_EXCEPTION((
-                    std::runtime_error{
-                        "Failed to connect to DBus system bus: "s + error_msg}));
+                                          std::runtime_error{
+                                              "Failed to connect to DBus system bus: "s + error_msg}));
             }
 
             return bus;
         }).get();
 }
+
+class LogindSessionLockObserver : public mir::frontend::SessionLockObserver
+{
+public:
+    LogindSessionLockObserver(
+        std::function<void()> const& on_lock,
+        std::function<void()> const& on_unlock)
+        : on_lock_{on_lock},
+          on_unlock_{on_unlock}
```

Comment:
As `LogindSessionLockObserver` is only used in this file, is there any need for an indirection through `std::function`? Can't `logind_session_set_locked_hint()`  be called directly?

Reply:
This^